netbox/netbox.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, script_args, script_config_data):
         # Script configuration.
         self.script_config = script_config_data
         self.api_url = self._config(["main", "api_url"])
+        self.api_token = self._config(["main", "api_token"], ignore_key_error=True)
```

Comment:
Why `ignore_key_error`? This should be `default={}` instead ignoring key error.

Reply:
Letting the default value be assigned when `api_token` was missing from `netbox.yml` was resulting a sys.exit() in _config() due to the `if value:` conditional evaluation on line 144-148 (that I removed/changed).

Looking into it.